deleting users and more

biel535 opened the pull request.

biel535 committed:
users add and form

biel535 committed:
feat(edit user): :fire: edit user option

Edit user data implemented on users table. Created userOptions, URLS method, useUser hook updated of editUser mutated funcion.

biel535 committed:
feat(delete user): :heavy_minus_sign: delete user feature added

To users table has been added functionality of deleting user. Also the table is more responsive and the "dostosuj kolumny" button has nice names instead of for example "firstName".

bykowskiolaf commented on src/hooks/useCompanies/index.tsx:
Please use prettier.

bykowskiolaf commented on src/pages/Me/index.tsx:
```diff
@@ -17,7 +17,8 @@ const Me = () => {
         Link to users
       </Link>
-    </>
+      <Link to="/usersTable">Link to users table</Link>
```
Change to /users

bykowskiolaf commented on src/pages/UserForm/index.tsx:
```diff
@@ -0,0 +1,339 @@
+export function UserForm({ user, isOpen, onOpenChange }: UserFormProps) {
+  const queryClient = useQueryClient();
+  const form = useForm<UserRequest>({
+    //resolver: zodResolver(formSchema),
```
Please enable form validation.

bykowskiolaf commented on src/pages/UserForm/index.tsx:
```diff
@@ -0,0 +1,339 @@
+  SelectValue
+} from '../../components/ui/select';
+
+const formSchema = z.object({
```
The messages should be in polish as this is visible by the user.

bykowskiolaf commented on src/pages/UserForm/index.tsx:
```diff
@@ -0,0 +1,339 @@
+    const workYears = new Date().getFullYear() - workStart.getFullYear();
+
+    if (user) {
+      editUser(data).then(() => toast.success('Pomyślnie zedytowano dane!'));
```
add `.catch(() => ...)`

bykowskiolaf commented on src/pages/UserForm/index.tsx:
```diff
@@ -0,0 +1,339 @@
+                                <SelectValue placeholder="Firma" />
+                              </SelectTrigger>
+                              <SelectContent>
+                                {companiesLoading ? (
```
Super, że sprawdzasz czy sa na pewno dane

bykowskiolaf commented on src/pages/Users/index.tsx:
```diff
@@ -2,7 +2,7 @@ import useUsers from '@/hooks/useUsers';
 const Users = () => {
   // Here we are using the hook 'useUsers' to fetch users from the backend. We are deconstructing the users element.
-  const { users } = useUsers();
+  const { users  } = useUsers();
```
Prettier

bykowskiolaf commented on src/hooks/useUsers/index.tsx:
```diff
@@ -1,20 +1,78 @@
+    isPending: deleteUserPending,
+    isError: deleteUserError
+     } = useMutation({
+    mutationFn: (userData: UserRequest): Promise<UserRequest> =>
```
This shouldnt take whole UserRequest object if all it uses is id. It should only take the id as an argument.

`mutationFn: (id: string): Promise<void> =>`

bykowskiolaf commented on src/pages/UsersTable/UserOptions.tsx:
```diff
@@ -0,0 +1,72 @@
+
+  const handleDeleteUser = async () => {
+    try {
+      await deleteUser({
```
instead of `try { await ...` use `deleteUser.then(() => ...).catch(() => ...)`

bykowskiolaf replied:
Also just call the function with `id` now

bykowskiolaf commented on src/pages/UsersTable/columns.tsx:
```diff
@@ -0,0 +1,100 @@
+
+import UserOptions from './UserOptions';
+
+const rolTranslation = {
```
`roleTranslation`

bykowskiolaf commented on src/pages/UsersTable/columns.tsx:
```diff
@@ -0,0 +1,100 @@
+  [Role.MENAGER]: 'Menager'
+};
+
+const conTranslation = {
```
`contractTranslation`, no need to shorten names like this

bykowskiolaf commented on src/pages/UsersTable/dataTable.tsx:
```diff
@@ -0,0 +1,180 @@
+    <div>
+      <div className="flex items-center py-4 g-4">
+        <Input
+          placeholder="Wyszukaj e-mail..."
```
It would be nice to make this table generic in the future so we can use it in other places as well.

bykowskiolaf commented on src/pages/UsersTable/dataTable.tsx:
```diff
@@ -0,0 +1,180 @@
+              })}
+          </DropdownMenuContent>
+        </DropdownMenu>
+        <Button onClick={() => setAddUserDialogOpen(true)} variant="outline">
```
Same here, this could be passed as a prop in the future

bykowskiolaf commented on src/pages/UsersTable/dataTable.tsx:
```diff
@@ -0,0 +1,180 @@
+                <TableCell
+                  colSpan={columns.length}
+                  className="h-24 text-center">
+                  No results.
```
`Brak wyników.`

bykowskiolaf commented on src/pages/UsersTable/page.tsx:
Prettier

bykowskiolaf commented on src/types/User.ts:
```diff
@@ -0,0 +1,46 @@
+import { CompanyDto } from './Company';
+
+export enum Role {
```
Mozna pomyslec nad zmiana na `UserRole`

bykowskiolaf commented on src/pages/UsersTable/columns.tsx:
```diff
@@ -0,0 +1,100 @@
+  [Contract.ZLECENIE]: 'Umowa zlecenie',
+  [Contract.PRACA]: 'Umowa o pracę'
+};
+const userFieldNames = {
```
Use field names from `User.ts`

bykowskiolaf commented on src/routes/_dashboard/usersTable.tsx:
Prettier

bykowskiolaf commented on src/routes/_dashboard/UserForm.tsx:
Prettier

biel535 committed:
fix(code prettier and more functional): :art: prettier, more functional many files

deleteUser as .then not try. DeleteUser -> deleting by  only id. ZodResolver is working. Form is prettier (button width etc).

bykowskiolaf commented on src/types/User.ts:
```diff
@@ -10,7 +10,7 @@ export enum Contract {
 }
 
-export interface User {
+export interface UserRole {
```
User should be called user, but the role inside should be called UserRole

bykowskiolaf commented on src/pages/UsersTable/page.tsx:
```diff
@@ -1,18 +1,15 @@
+import { columns } from './columns';
+import { DataTable } from './dataTable';
 
 export default function DemoPage() {
```
Change name

biel535 committed:
fix(code prettier and more functional): :art: just cleaning

changing demoPage to TableComponent, deleting not used imports and props

biel535 committed:
refactor: :art: changed project structure

bykowskiolaf reviewed (approved):
Superb!
